File: 905.Sort-Array-By-Parity/SortArrayByParity.go
// 905. Sort Array By Parity
// https://leetcode.com/problems/sort-array-by-parity/

package main

func sortArrayByParity(A []int) []int {
	b := make([]int, len(A))
	s := 0
	e := len(A) - 1

	for _, v := range A {
		if v%2 == 0 {
			b[s] = v
			s += 1
		} else {
			b[e] = v
			e -= 1
		}
	}
	return b
}

File: 🙂easy/0905.Sort-Array-By-Parity/sort_array_by_parity.py
# 905. Sort Array By Parity
# https://leetcode.com/problems/sort-array-by-parity/description/

from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        i, j = 0, len(nums) - 1
        while i < j:
            if nums[i] % 2 != 0:
                nums[i], nums[j] = nums[j], nums[i]
                j -= 1
            else:
                i += 1
        return nums
